src/protocol.cpp

```diff
@@ -211,10 +217,23 @@ std::string serviceFlagToStr(const uint64_t mask, const int bit)
     stream.imbue(std::locale::classic());
     stream << "UNKNOWN[";
     if (bit < 8) {
-        stream << mask;
+        stream << service_flag;
```

Comment:
Why is this branch needed? 2^7 or similar should read just fine and help with consistency, no?

Reply:
It was like this in the original code and I am trying to limit the amount of changes. Especially this would be user-visible change. Otherwise I agree that for consistency we may just print `2^N`.

I do not know why it was done like that. Maybe with the assumption that e.g. `64` is more human-readable than `2^6`, but `524288` is *not* more human-readable than `2^19`.

It was introduced in
https://github.com/bitcoin/bitcoin/commit/df77de8c2#diff-a459a939be4fe065dd5f64ab60bb30d0L842
@luke-jr ?

Reply:
No, it predates that commit too.